docs/src/architecture/accounts.md

```diff
@@ -8,44 +8,51 @@ An account is an entity which holds assets and defines rules of how these assets
 In the above:
 
 * **Account ID** is a unique identifier (8 bytes) of an account which does not change throughout its lifetime. It also specifies the type of the underlying account. 
-* **Storage** is user-defined data which can be stored in an account. The data is stored as key-value map which maps 256-bit keys to 256-bit values.
-* **Nonce** is the account's nonce increments every time the account changes its state. 
+* **Storage** is user-defined data which can be stored in an account. The data can be stored in a simple `SparseMerkleTree` (256 Word slots), or if additional capacity is needed in a `MerkleStore` allowing the user to store any Merkle structures users need.
+* **Nonce** is a counter that increments every time the account changes its state. 
 * **Vault** is collection of assets stored in an account. In this collection all fungible assets with the same ID are grouped together, and all non-fungible assets are represented as individual entries.
 * **Code** is a collection of functions which define an external interface for an account. The internal state of the account can only be modified by invoking these functions.
 
 Functions exposed by the account have the following properties:
 
-* Functions are actually roots of Miden program MASTs (i.e., 32-byte hash). Thus, function identifier is a commitment to the code which is executed when a function is invoked.
+* Functions are actually roots of [Miden program MASTs](https://wiki.polygon.technology/docs/miden/user_docs/assembly/main) (i.e., 32-byte hash). Thus, function identifier is a commitment to the code which is executed when a function is invoked.
 * Only account functions have mutable access to an account's storage and vault. Said another way, the only way to modify an account's internal state is through one of account's functions.
 * Account functions can take parameters and can create new notes.
 
-## Types of Accounts
-There are four types of Accounts in Miden. The first two bits of the Account ID specify the type of the account.
+## Account lifecycle
+For an account to exist it must be present in the [Account DB](https://0xpolygonmiden.github.io/miden-base/architecture/state.html#account-database) kept by the Miden Node(s). However, new accounts can be created locally by users using a wallet.
 
-### Regular Account with updatable code
+* Alice grinds a new Account ID according to the account types (see below) using a wallet
```

Comment:
I'd probably add an intro sentence before diving into this example.

Reply:
good point, I added an intro. However, also the 2 sentences above the bullet list that describe the process could be seen as intro.